### PR TITLE
Auto Hide Menu Bar Settings

### PR DESCRIPTION
**Pull request by shirshak55**
This PR gives ability to hide menubar. I have never used them and I think many users will never use them too.

Users can always press Alt to reveal those menus just like in firefox or vscode.

Maybe it is better to document in the HTTP toolkit website. Also maybe we can add it in the settings of the renderer process.

For future:

Before:
Screenshot from 2021-07-29 08-47-22

After:
Screenshot from 2021-07-29 08-47-18

### CODE REVIEW

**Comment by shirshak55** [comment]
I don't know how we persist in settings. May be its time to use basic setting storage like electron store?

File: src/index.ts [hunk 1]
```diff
@@ -566,4 +566,12 @@ if (!amMainInstance) {
             app.quit();
         }
     });
+}
+
+
+export function autoHideMenuBar(status: boolean){
+    windows.forEach((window) =>
+        window.setAutoHideMenuBar(status || false)
```
Comment (pimterry) [inline src/index.ts:574]: I don't think you need `|| false` here: TypeScript will enforce that the status is a boolean (so not undefined or null, always just `true` or `false`) so AFAICT this isn't doing anything.

File: src/menu.ts [hunk 2]
```diff
@@ -27,6 +28,9 @@ const menuTemplate: MenuItemConstructorOptions[] = [
             { role: 'reload' },
             { role: 'forceReload' },
             { role: 'toggleDevTools' },
+            { type:"checkbox","label":"Toggle Menu Bar", click(event:any){
```
Comment (pimterry) [inline src/menu.ts:31]: I don't think you need `any` here. If you remove that entirely, TypeScript should be able to infer the event type correctly.

Nitpicking, but: it's also be nice to add some spaces in here, to make the codestyle match the rest of this file.

**Review by pimterry — COMMENTED** [review]
Two very minor points, but otherwise yep this looks great to me :+1:. Persisting it would be good too, Electron-Store looks like a good option for that to me.

**Comment by pimterry** [comment]
Thanks! I've done some testing and tweaked it a little:

* I've refactored the structure of the previously existing menu into a function. That allows us to move all of the logic related to this functionality into the menu file (to avoid an index <-> menu circular dependency) and should make it easier to manage and extend the menu for other features in future.
* There was a small bug, because `checked` wasn't set, so the menu option didn't show as checked at startup even if it was enabled, and the first click after startup wouldn't disable it.
* I've hidden it on Mac, because it doesn't make sense (the menu bar is _always_ visible as part of the OS UI) and so APIs like `setMenuBarVisibility` are only supported on Linux & Windows.

Otherwise I think this is a nice little improvement, thanks! Merging now :+1:

**Comment by shirshak55** [comment]
@pimterry how are updates distributed. I downloaded HTTP toolkit to see that option but I can't find it.

**Comment by pimterry** [comment]
It's not published yet. At some point soon I'll bump the version to create a new [release](https://github.com/httptoolkit/httptoolkit-desktop/releases), then manually test the resulting builds on every platform to make sure nothing's broken, then update the version that's used for download on the web site [here](https://github.com/httptoolkit/httptoolkit.tech/blob/master/gatsby-config.js#L10).

If you want to use it yourself ASAP in the meantime you can download complete builds directly from every CI build, there's separate linux, mac & windows distributable zips attached to all of them, for example: https://github.com/httptoolkit/httptoolkit-desktop/actions/runs/1090764188

**Comment by shirshak55** [comment]
thanks :) I see it requires release tags.

**Comment by pimterry** [comment]
Just to let you know, this is now released as v1.4.4, available from [releases](https://github.com/httptoolkit/httptoolkit-desktop/releases) here but also as the new latest version on [httptoolkit.tech](https://httptoolkit.tech/). Thanks again for contributing this great feature!

**Pull request merged**